includes/html/pages/device/health.inc.php

```diff
@@ -18,215 +39,78 @@
-$bitrate = Sensor::where('sensor_class', 'bitrate')->where('device_id', $device['device_id'])->count();
-
 unset($datas);
 $datas[] = 'overview';
-if ($processor) {
+
+if (Processor::where('device_id', $device['device_id'])->count()) {
     $datas[] = 'processor';
 }
 
 if ($qfp) {
     $datas[] = 'qfp';
 }
 
-if ($mempools) {
+if (Mempool::where('device_id', $device['device_id'])->count()) {
     $datas[] = 'mempool';
 }
 
-if ($storage) {
+if (Storage::where('device_id', $device['device_id'])->count()) {
     $datas[] = 'storage';
 }
 
-if ($diskio) {
+if (DiskIo::where('device_id', $device['device_id'])->count()) {
     $datas[] = 'diskio';
 }
 
-if ($charge) {
-    $datas[] = 'charge';
-}
-
-if ($temperatures) {
-    $datas[] = 'temperature';
-}
-
-if ($humidity) {
-    $datas[] = 'humidity';
-}
-
-if ($fans) {
-    $datas[] = 'fanspeed';
-}
-
-if ($volts) {
-    $datas[] = 'voltage';
-}
-
-if ($freqs) {
-    $datas[] = 'frequency';
-}
-
-if ($runtime) {
-    $datas[] = 'runtime';
-}
-
-if ($current) {
-    $datas[] = 'current';
-}
-
-if ($power) {
-    $datas[] = 'power';
-}
-
-if ($power_consumed) {
-    $datas[] = 'power_consumed';
-}
-
-if ($power_factor) {
-    $datas[] = 'power_factor';
-}
-
-if ($dBm) {
-    $datas[] = 'dbm';
-}
-
-if ($states) {
-    $datas[] = 'state';
-}
-
-if ($count) {
-    $datas[] = 'count';
-}
-
-if ($load) {
-    $datas[] = 'load';
-}
-
-if ($signal) {
-    $datas[] = 'signal';
-}
-
-if ($tv_signal) {
-    $datas[] = 'tv_signal';
-}
-
-if ($bitrate) {
-    $datas[] = 'bitrate';
-}
-
-if ($airflow) {
-    $datas[] = 'airflow';
-}
-
-if ($snr) {
-    $datas[] = 'snr';
-}
-
-if ($pressure) {
-    $datas[] = 'pressure';
-}
-
-if ($cooling) {
-    $datas[] = 'cooling';
-}
-
-if ($delay) {
-    $datas[] = 'delay';
-}
-
-if ($quality_factor) {
-    $datas[] = 'quality_factor';
-}
-
-if ($chromatic_dispersion) {
-    $datas[] = 'chromatic_dispersion';
-}
-
-if ($ber) {
-    $datas[] = 'ber';
-}
-
-if ($eer) {
-    $datas[] = 'eer';
-}
-
-if ($waterflow) {
-    $datas[] = 'waterflow';
-}
-
-if ($loss) {
-    $datas[] = 'loss';
-}
+$sensors = [
+    'airflow', 'ber', 'bitrate', 'charge', 'chromatic_dispersion', 'cooling', 'count', 'current', 'dBm', 'delay', 'eer',
+    'fanspeed', 'frequency', 'humidity', 'load', 'loss', 'percent', 'power', 'power_consumed', 'power_factor', 'pressure',
+    'runtime', 'signal', 'snr', 'state', 'temperature', 'tv_signal', 'voltage', 'waterflow', 'quality_factor',
+];
 
-if ($percent) {
-    $datas[] = 'percent';
+foreach ($sensors as $sensor_name) {
+    if (Sensor::where('sensor_class', $sensor_name)->where('device_id', $device['device_id'])->count()) {
+        //strtolower because 'dBm - dbm' difference
+        $datas[] = strtolower($sensor_name);
+    }
 }
 
 $type_text['overview'] = 'Overview';
-$type_text['charge'] = 'Battery Charge';
-$type_text['temperature'] = 'Temperature';
-$type_text['humidity'] = 'Humidity';
```

Comment:
All of these already exists in the translation files https://github.com/librenms/librenms/blob/2d57e49cafffd2982c393ef0ffefab763b387b8c/resources/lang/en/sensors.php#L33
So you could use that

Reply:
@Jellyfrog 
done